app/features/vods/VodRepository.server.ts

```diff
@@ -1,5 +1,276 @@
+		);
+	}
+	const result = await query
+		.groupBy("Video.id")
+		.orderBy("Video.youtubeDate", "desc")
+		.limit(limit)
+		.execute();
+
+	const vods = result.map((value) => {
+		const { playerNames, players, ...vod } = value;
+		const playerNamesArray: string[] = playerNames as string[];
+		const playersArray = players;
+		return {
+			...vod,
+			pov: playerNamesArray[0] ?? playersArray[0],
+		};
+	});
+	return vods;
+}
+
+export async function findVodById(id: Tables["Video"]["id"]) {
+	const video_query = db
+		.selectFrom("Video")
+		.select([
+			"id",
+			"title",
+			"youtubeDate",
+			"youtubeId",
+			"type",
+			"submitterUserId",
+		])
+		.where("Video.id", "=", id);
+
+	const video = await video_query.executeTakeFirst();
+
+	if (video) {
+		const video_match_query = db
+			.selectFrom("VideoMatch")
+			.select([
+				"VideoMatch.id",
+				"VideoMatch.mode",
+				"VideoMatch.stageId",
+				"VideoMatch.startsAt",
+			])
+			.leftJoin(
+				"VideoMatchPlayer",
+				"VideoMatch.id",
+				"VideoMatchPlayer.videoMatchId",
+			)
+			.leftJoin("User", "VideoMatchPlayer.playerUserId", "User.id")
+			.select(selectPlayers())
+			.select(({ fn }) => [
+				fn
+					.agg("json_group_array", ["VideoMatchPlayer.weaponSplId"])
+					.$castTo<MainWeaponId[]>()
+					.as("weapons"),
+				fn
+					.agg("json_group_array", ["VideoMatchPlayer.playerName"])
+					.as("playerNames"),
+			])
+			.where("VideoMatch.videoId", "=", id)
+			.groupBy("VideoMatch.id")
+			.orderBy("VideoMatch.startsAt", "asc")
+			.orderBy("VideoMatchPlayer.player", "asc");
+
+		const matches = await video_match_query.execute();
+
+		return {
+			...video,
+			pov: resolvePov(matches),
+			matches: R.map(matches, R.omit(["players", "playerNames"])),
+		};
+	}
+	return null;
+}
+
+function resolvePov(matches: any): Vod["pov"] {
+	for (const match of matches) {
+		if (match.playerNames.length > 0) {
+			return match.playerNames[0];
+		}
+
+		if (match.players.length > 0) {
+			return match.players[0];
+		}
+	}
+
+	return;
+}
+
+export async function updateVodByReplacing(
+	args: VideoBeingAdded & {
+		submitterUserId: number;
+		isValidated: boolean;
+		id: number;
+	},
+) {
+	return createVod(args);
+}
+
+export async function createVod(
+	args: VideoBeingAdded & {
+		submitterUserId: number;
+		isValidated: boolean;
+		id?: number;
+	},
+) {
+	const youtubeId = extractYoutubeIdFromVideoUrl(args.youtubeUrl);
+	invariant(youtubeId, "Invalid YouTube URL");
+	return db.transaction().execute(async (trx) => {
+		let videoId: number;
+		const video = {
+			title: args.title,
+			type: args.type,
+			youtubeDate: dayMonthYearToDatabaseTimestamp(args.date),
+			eventId: args.eventId ?? null,
+			youtubeId,
+			submitterUserId: args.submitterUserId,
+			validatedAt: args.isValidated
+				? dateToDatabaseTimestamp(new Date())
+				: null,
+		};
+		if (args.id) {
+			await trx
+				.deleteFrom("VideoMatch")
+				.where("videoId", "=", args.id)
+				.executeTakeFirstOrThrow();
```

Comment:
```suggestion
				.execute();
```

You are not returning any value